modules/geo-layers/src/tile-layer/utils.js

```diff
@@ -177,11 +177,12 @@ export function getTileIndices({
   extent,
   tileSize = TILE_SIZE,
   modelMatrix,
-  modelMatrixInverse
+  modelMatrixInverse,
+  tileOffset = 0
 }) {
   let z = viewport.isGeospatial
     ? Math.round(viewport.zoom + Math.log2(TILE_SIZE / tileSize))
-    : Math.ceil(viewport.zoom);
+    : Math.ceil(viewport.zoom) + tileOffset;
```

Comment:
Another option here would be `Math.ceil(viewport.zoom + tileOffset)` which is less error-prone for users (since it would also produce a valid `z` value) but would be less interpretable since we apply the `ceil` operation.